src/silx/io/dictdumplinks/_vds.py

```diff
@@ -59,6 +61,37 @@ def tolink(self, source: DataUrl) -> h5py.VirtualLayout:
         return vds_layout
 
 
+class VdsUrlsModelV1(Hdf5LinkModel, arbitrary_types_allowed=True):
+    dictdump_schema: Literal["vds_urls_v1"]
+    source_shape: tuple[int, ...]
+    source_dtype: (
+        Any  # DTypeLike gives pydantic.errors.PydanticUserError on Python < 3.12.
+    )
+    sources: DataUrl | list[DataUrl]
+
+    @field_validator("sources", mode="before")
+    @classmethod
+    def as_dataurl(
+        cls, value: DataUrl | list[DataUrl] | str | Sequence[str]
+    ) -> RawDsetIndex:
+        if isinstance(value, str):
+            return DataUrl(value)
+        if isinstance(value, (list, tuple)) and isinstance(value[0], str):
+            return [DataUrl(s) for s in value]
+        return value
+
+    def tolink(self, source: DataUrl) -> h5py.VirtualLayout:
+        if isinstance(self.sources, DataUrl):
+            target_schema = hdf5_url_to_vds_schema(
+                source, self.sources, self.source_shape, self.source_dtype
+            )
+        else:
+            target_schema = hdf5_urls_to_vds_schema(
+                source, self.sources, self.source_shape, self.source_dtype
+            )
```

Comment:
Passing `self.source_*` as `target_*` args is misleading IMO, best to make it explicit:

```suggestion
            target_schema = hdf5_urls_to_vds_schema(
                source,
                targets=self.sources,
                target_shape=self.source_shape,
                target_dtype=self.source_dtype,
            )
```

or to use `source` only for "target" (since it is used this way for VDS in h5py) and use something else for `source` when it refer to the URL of the link (e.g,`link` or `link_url`)